src/absil/il.fs

```diff
@@ -2016,6 +2016,17 @@ type ILTypeDef(name: string, attributes: TypeAttributes, layout: ILTypeDefLayout
     new (name, attributes, layout, implements, genericParams, extends, methods, nestedTypes, fields, methodImpls, events, properties, securityDecls, customAttrs) =  
        ILTypeDef (name, attributes, layout, implements, genericParams, extends, methods, nestedTypes, fields, methodImpls, events, properties, storeILSecurityDecls securityDecls, storeILCustomAttrs customAttrs, NoMetadataIdx)
 
+    member val CachedFilteredCustomAttributes = None with get, set
```

Comment:
I'm skeptical we should be adding an extra field to ILTypeDef for such a very corner case feature.  @AviAvni worked  very hard to reduce the size of ILTypeDef in 2018....  Many, many of these get instantiated.

Why not just filter on each check through the attributes?

Reply:
I think I agree with you that we shouldn't add something like this to `ILTypeDef`. I think it makes the API confusing.

Reply:
Ah I see. It's not just the API. We are also storing more data here.